### ISSUE TITLE
Share files in LAN and Internet

### ISSUE THREAD

**Issue by davidak** [elementary/files#1640]
## Problem
I want to send a file to a friend.

I might visit him currently (so my notebook is in his LAN) or more likely in the pandemic, we are in 2 different houses, so the internet has to be used.

This sounds like a problem simple to use, but the popular solutions are horrible!

https://lifehacker.com/what-s-the-best-way-to-share-large-files-with-friends-1486810038
https://www.makeuseof.com/tag/5-super-easy-tools-share-files-friends-devices/

## Proposal
Apple has a very user friendly design here with [AirDrop](https://support.apple.com/en-us/HT203106). The problem is, it works only on Apple products.

It would be great to add such a feature to elementary Files, but it should also be compatible with other possible implementation. Maybe you can collaborate with GNOME here, so they have it too in their files app.

[ShareDrop](https://www.sharedrop.io/) has implemented such a tool in the browser using WebRTC. It could be used as fallback.

So i expect to be able to right-click a file, select "Send to" -> select either a person in LAN or "Share with Link", which gives a ShareDrop link the other person can use do receive the files.

There should also be a stand-alone desktop program, that might adds itself to the status bar. Someone else could implement it. It would be great to be platform independent, so Mac, Windows and *BSD users can also participate in this.

That program could be developed with GNOMEs libhandy, so it can be also used and integrate itself into mobile Linux (smartphones, tablets, smartwatches, ...).

It should really be a user friendly, platform independent solution to share files.

## Prior Art

- [AirDrop](https://support.apple.com/en-us/HT203106) by Apple
- [ShareDrop](https://www.sharedrop.io) (Free Software, works in Browser, using WebRTC)
- [instant.io](https://instant.io/) (Free Software, works in Browser, using WebTorrent)
- [croc](https://github.com/schollz/croc) and [magic-wormhole](https://github.com/magic-wormhole/magic-wormhole) (Free Software, CLI tools)

**Comment by jeremypw** [comment]
Probably best to limit the issue initially to sharing files on the same network? Sharing over the internet securely without a third party is more demanding.  Presumably all the above solutions require both devices to have the same software solution installed? So are we talking about sharing to other elementaryos devices or other Linux devices or to other OSes?

**Comment by davidak** [comment]
>Presumably all the above solutions require both devices to have the same software solution installed?

different software solutions can use the same _protocol_, so **no**. like there are different bittorrent or e-mail clients

>So are we talking about sharing to other elementaryos devices or other Linux devices or to other OSes?

the end goal would be to share files between every device where you have files. computers, smartphones, smartwatches, toasters

the free software developer community should create a standard protocol that is so popular android will include it by default. and elementary OS can lead with a good integration directly in the files program (that has to happen at a late stage where the protocol and a robust library already exist)

it's a big goal, i know. but we can start small. i currently don't have the resources to initiate such effort, so for now i just shared the idea

it's crazy that we still don't have an easy option to share files between two devices

**Comment by jeremypw** [comment]
> different software solutions can use the same protocol, so no. 

You are right of course, my language was too loose.  But is there a suitable common protocol to target?  Until there is I am not sure how to progress this.

**Comment by jeremypw** [comment]
This probably involves having elementary come with suitable server client software oob and an easy way to establish a (secure) link over the internet.  Not trivial and not entirely within the province of Files. I am inclined to transfer this to triage for consideration.

**Comment by pedropaulosuzuki** [comment]
About sharing files while being close to someone, but not on the same network, there is [WiFi Direct](https://en.wikipedia.org/wiki/Wi-Fi_Direct), which has been used on phones for more than a decade for file sharing (and also for other stuff, like [screencasting](https://en.wikipedia.org/wiki/Miracast)). As there is already built-in support for Bluetooth file sharing (which I use all the time for small file transfers), it would be nice to be able to do the same with WiFi Direct, [which can achieve much higher speeds](https://youtu.be/Ue_AEHeuiL8) - see: [1.46 GB on roughly a minute](https://youtu.be/uXznlcTpzDM).
